activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/helper/TaskVariableCopier.java

```diff
@@ -0,0 +1,25 @@
+package org.activiti.engine.impl.bpmn.helper;
+
+import org.activiti.engine.impl.persistence.entity.TaskEntity;
+
+public class TaskVariableCopier {
+
+    public static void copyVariablesIntoTaskLocal(TaskEntity task){
+
+        //TODO: would like to filter which variables copy much as with subProcesses
+
+        task.setVariablesLocal(task.getVariables());
+
+
+    }
+
+    public static void copyVariablesOutFromTaskLocal(TaskEntity task){
+
+        //TODO: would like to filter which variables copy much as with subProcesses
+
+        //provided not a standalone task
```

Comment:
@ryandawsonuk is this not redundant? the Behaviour will not be called unless the task creation happens from a process instance.

Reply:
the check is for completion as that goes through CompleteTaskCmd and standalone tasks also use that